### ISSUE TITLE
Feature: Show line numbers

### ISSUE THREAD

**Issue by agscripter** [facebook/lexical#6532]
## Description
Show line numbers in the number to guide author and readers.

## Impact
It improves document readability and user experience. It helps authors to understand how many lines they have written and readers to resume the reading or to share content with others.

**Comment by Kingscliq** [comment]
Hi @agscripter this issue is not so clear, is there a reference we can use to clarify the issue more, maybe I can hop on it

**Comment by agscripter** [comment]
Hi @Kingscliq ,

My request is to have something similar to Quip, we can enable "Line numbers", like in this screenshot below:

**Observe that we have numbers on the left side of each row.**
<img width="1005" alt="Screenshot 2024-10-22 at 6 00 38 PM" src="https://github.com/user-attachments/assets/fab3f3b8-80aa-4f24-8d9f-55a34aeef7dc">

**Comment by etrepum** [comment]
This sort of problem isn't so easy to solve in a general and performant way because the layout in HTML documents is fluid. If you resize the browser or change the font then the lines will change.

**Comment by Kingscliq** [comment]
> This sort of problem isn't so easy to solve in a general and performant way because the layout in HTML documents is fluid. If you resize the browser or change the font then the lines will change.

Yeah I agree, so you have to recalculate the line number function each time the browser window is resized

**Comment by Kingscliq** [comment]
Hi @etrepum I am coming from MLH fellowship, and looking forward to make my first contribution, but I think I am stuck In finding a good first issue, most of the issues are either taken or has been fixed in an open PR. Do you have any recommendation for me, even its a documentation issue that I can update. I could see a couple of ways I can improve the copys on the docs on the lexical website

**Comment by etrepum** [comment]
Documentation and examples are always appreciated, and there's a lot of missing content there! In many cases there won't be a corresponding issue for simpler tasks just because nobody is thinking about them, or they get closed quickly after they're created. One documentation issue that has come up a few times recently is that the editor configuration has an html property that is not documented, which lets you basically add importDOM and exportDOM implementations without using node replacement. That might be worth looking into, especially if you plan to add an example.

**Comment by Kingscliq** [comment]
> Documentation and examples are always appreciated, and there's a lot of missing content there! In many cases there won't be a corresponding issue for simpler tasks just because nobody is thinking about them, or they get closed quickly after they're created. One documentation issue that has come up a few times recently is that the editor configuration has an html property that is not documented, which lets you basically add importDOM and exportDOM implementations without using node replacement. That might be worth looking into, especially if you plan to add an example.

Thanks @etrepum, is there a link to the issue or do I need to create one, also references to files I can see that implementation would be very helpful 🙏

**Comment by etrepum** [comment]
There's no existing issue that I'm aware of. I'm referring to the html property (with the HTMLConfig type) of CreateEditorArgs which is used for createEditor and LexicalComposer's initialConfig.

**Comment by Kingscliq** [comment]
> There's no existing issue that I'm aware of. I'm referring to the html property (with the HTMLConfig type) of CreateEditorArgs which is used for createEditor and LexicalComposer's initialConfig.

Okay, thanks. I will definitely check this out. Can I chat with you privately on Discord or in the general channel in case I need more directions and guidance on this or other potential issues I could work on?
